### ISSUE TITLE
--list-different is reporting files are different (when they aren't)

### ISSUE THREAD

**Issue by wayneashleyberry** [prettier/prettier#1971]
I've run `prettier --write '**/*.{css,ts}'` on my code base and committed the changes. Using `prettier -l '**/*.{css,ts}'` afterwards returns a `1` exit code causing my CI to fail.

Both `ts` and `css` files are being reporting as being incorrectly formatted - but running prettier on them again results in no diff.

Adding the write flag changes the output, in that no files get listed - but the exit code is still non-zero. `prettier --write --list-different '**/*.{ts,css}'`

Reproduction can be found [here](https://github.com/wayneashleyberry/prettier-test-repo).

---

Running `--debug-check` shows the following, which interestingly enough reports unexpected tokens for the problematic files.

```sh
~/prettier-test-repo master
❯ prettier --debug-check '**/*.{ts,css}'
color.directive.ts
color.directive.ts: SyntaxError: Unexpected token (17:11)
  15 | })
  16 | export class ColorPickerDirective implements AfterViewInit, OnChanges {
> 17 |   private _element: any;
     |           ^
  18 |
  19 |   @Input() colorValue: string;
  20 |   @Output() colorValueChanged: EventEmitter<any> = new EventEmitter();
index.ts
lib/foo.ts
public/style.css
public/style.css: SyntaxError: Unexpected token, expected ; (1:6)
> 1 | body {
    |      ^
  2 |   font-family: sans-serif;
  3 | }
  4 |
```

---

This screenshot might help as well :)

<img width="1268" alt="screen shot 2017-06-05 at 12 28 21 pm" src="https://cloud.githubusercontent.com/assets/727262/26780530/79f2b5b6-49ea-11e7-8a80-6bbd56884ef7.png">

**Comment by wayneashleyberry** [comment]
This screenshot might help as well :)

<img width="1268" alt="screen shot 2017-06-05 at 12 28 21 pm" src="https://cloud.githubusercontent.com/assets/727262/26780530/79f2b5b6-49ea-11e7-8a80-6bbd56884ef7.png">

**Comment by wayneashleyberry** [comment]
Looks like I wasn't setting `--parser` :(

**Comment by azz** [comment]
I think this is actually a bug. I think prettier is supposed to be using the extension to determine the parser when using a glob.

**Comment by philipp-spiess** [comment]
It seems like `--write` will change the parser to `postcss`, when a `.s?css` file is being parsed but this behavior is not present when using `--list-different`.

**Comment by vjeux** [comment]
I just published 1.4.3 which fixes it. Thanks for reporting it!
